stock_picking_actual_date/models/stock_move.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    actual_date = fields.Date(
+        compute="_compute_actual_date",
+        store=True,
+    )
+
+    def _get_timezone(self):
+        return self.env.context.get("tz") or self.env.user.tz or "UTC"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get("actual_date"):
+                self = self.with_context(actual_date=vals["actual_date"])
+        return super().create(vals_list)
```

Comment:
A bit tedious but somethings like this?
```suggestion
    moves = self.env["stock.move"]
    vals_list_with_actual_date = []
    vals_list_without_actual_date = []
    for vals in vals_list:
        if vals.get("actual_date"):
            vals_list_with_actual_date.append(vals)
        else:
            vals_list_without_actual_date.append(vals)
    if vals_list_with_actual_date:
        for vals in vals_list_with_actual_date:
            move = super().with_context(actual_date=vals["actual_date"]).create(vals)
            moves += move
    if vals_list_without_actual_date:
        moves += super().create(vals_list_without_actual_date)
    return moves
```